hack/benchmark/core/benchmark/strategy/util.go

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package strategy provides benchmark strategy
+package strategy
+
+import (
+	"context"
+
+	"github.com/vdaas/vald/hack/benchmark/internal/assets"
+	"github.com/vdaas/vald/hack/benchmark/internal/core"
+	"github.com/vdaas/vald/internal/errors"
+)
+
+const (
+	bulkInsertCnt = 1000
+)
+
+func wrapErrors(errs []error) (wrapped error) {
+	for _, err := range errs {
+		if err != nil {
+			if wrapped == nil {
+				wrapped = err
+			} else {
+				wrapped = errors.Wrap(wrapped, err.Error())
+			}
+		}
+	}
+	return
+}
+
+func insertAndCreateIndex32(ctx context.Context, c core.Core32, dataset assets.Dataset) (ids []uint, err error) {
+	train := dataset.Train()
+	ids = make([]uint, 0, len(train)*bulkInsertCnt)
+
+	for i := 0; i < bulkInsertCnt; i++ {
+		inserted, errs := c.BulkInsert(train)
+		err = wrapErrors(errs)
+		if err != nil {
+			return nil, err
+		}
+		ids = append(ids, inserted...)
+	}
+
+	err = c.CreateIndex(uint32((len(train) * bulkInsertCnt) / 100))
```

Comment:
@kmrmt 
If the number of the train is large, `CreateIndex` often fails. 
So Can I use 10000 pool size ?.

Reply:
you mean often fails on your local PC?

Reply:
No, it is not. `CreateIndex` often fails on remote PC.

Reply:
does the PC have enough memory?
In production vald cluster usually use 100000 pool size for create indexing it actually works

Reply:
Sorry. The test exits, not the failure.:bow:

Reply:
oh, I will check it again.

Reply:
Since the test time is 1 second by default, so benchmark exit...:thinking:

Reply:
why don't you change the timeout of benchmark